src/main/java/org/linksharing/server/http/ApiRestController.java

```diff
@@ -69,18 +73,38 @@ ResponseEntity<UserProfileDetails> updateProfileDetails(Principal user,
     }
 
     @GetMapping("/profile/picture")
-    ResponseEntity<?> getProfilePicture(Principal user) throws FileNotFoundException {
+    ResponseEntity<?> getProfilePicture(Principal user) throws IOException {
 
-        File img = new File("../image.jpg");
+        UserProfileDetails userProfile = profileRepository.findByEmail(user.getName());
+        String imageUrl;
+
+        if (userProfile.getImageUrl() != null) {
+            imageUrl = userProfile.getImageUrl();
+        } else {
+            imageUrl = "src/main/resources/static/placeholder.jpg";
+        }
+
+        File profilePicture = new File(imageUrl);
         return ResponseEntity
                 .ok()
                 .contentType(IMAGE_JPEG)
-                .body(new InputStreamResource(new FileInputStream(img)));
+                .body(new InputStreamResource(new FileInputStream(profilePicture)));
     }
 
     @PostMapping("/profile/picture")
-    ResponseEntity<?> updateProfilePicture(Principal user) {
-        return null;
+    ResponseEntity<UserProfileDetails> updateProfilePicture(Principal user, @RequestParam("file") MultipartFile file) throws IOException {
+        UserProfileDetails userProfile = profileRepository.findByEmail(user.getName());
+
+        String[] fileNameParts = file.getOriginalFilename().split("\\.");
+        String fileExtension = "." + fileNameParts[fileNameParts.length - 1];
+
+        Path newFileName = Paths.get("D:\\Java_Projects\\link-sharing-app\\pics", userProfile.getEmail() + fileExtension);
```

Comment:
Does this folder exist on my local machine or a server for deployment?

Reply:
No, it's not.
So the choosen place is correct, but path should be relative insted of absolute?

Reply:
Yes.

Did you create the `pics` directory starting the app?